tgui/packages/tgui/interfaces/Map.tsx

```diff
@@ -0,0 +1,177 @@
+import { useBackend, useLocalState } from '../backend';
+import { Tabs, Slider, Section, NoticeBox, Table } from '../components';
+import { NtosWindow } from '../layouts';
+
+export type MapData = {
+  map_image: any; // base64 icon
+  user_x: number;
+  user_y: number;
+  user_z: number;
+  station_levels: number[];
+  z_override: number;
+};
+
+export const Map = (props, context) => {
+  const { act, data } = useBackend<MapData>(context);
+
+  const [minimapZoom, setMinimapZoom] = useLocalState<number>(
+    context,
+    `minimapZoom`,
+    100
+  );
+
+  const [showLegend, setShowLegend] = useLocalState<boolean>(
+    context,
+    `showLegend`,
+    false
+  );
+
+  const map_size = 255;
+  const zoom_mod = minimapZoom / 100.0;
+
+  return (
+    <NtosWindow resizable>
+      <NtosWindow.Content scrollable>
+        <Section title="Map Program">
+          <Tabs>
+            <Tabs.Tab>Levels: </Tabs.Tab>
+            {data.station_levels?.map((station_level) => (
+              <Tabs.Tab
+                key={station_level}
+                width="50px"
+                backgroundColor={
+                  data.z_override === station_level ? '#4972a1' : null
+                }
+                icon={data.user_z === station_level ? 'user' : 'minus'}
+                onClick={() =>
+                  act('z_override', { z_override: station_level })
+                }>
+                {station_level}
+              </Tabs.Tab>
+            ))}
+            {data.z_override ? (
+              <Tabs.Tab
+                icon="filter-circle-xmark"
+                onClick={() => act('z_override', { z_override: 0 })}>
+                Clear Override
+              </Tabs.Tab>
+            ) : (
+              ''
+            )}
+            <Tabs.Tab
+              icon="fa-circle-question"
+              onClick={() => setShowLegend(!showLegend)}>
+              {showLegend ? 'Hide Legend' : 'Show Legend'}
+            </Tabs.Tab>
+          </Tabs>
+          {/*
+          #define HOLOMAP_AREACOLOR_COMMAND     "#386d8099"
+          #define HOLOMAP_AREACOLOR_SECURITY    "#ae121299"
+          #define HOLOMAP_AREACOLOR_MEDICAL     "#6f9e00c2"
+          #define HOLOMAP_AREACOLOR_SCIENCE     "#A154A699"
+          #define HOLOMAP_AREACOLOR_ENGINEERING "#F1C23199"
+          #define HOLOMAP_AREACOLOR_OPERATIONS  "#E06F0099"
+          #define HOLOMAP_AREACOLOR_HALLWAYS    "#ffffffa5"
+          #define HOLOMAP_AREACOLOR_DOCK        "#0000FFCC"
+          #define HOLOMAP_AREACOLOR_HANGAR      "#777777"
+          #define HOLOMAP_AREACOLOR_CIVILIAN    "#5bc1c199"
```

Comment:
Would it be possible to define the area color on the /area and pass that along to the ui instead of having to define it in multiple places (code/ui)

Reply:
It would be possible, sure, I will look into this later

Reply:
addressed